ios/Capacitor/Capacitor/CapacitorBridge.swift

```diff
@@ -172,6 +174,12 @@ internal class CapacitorBridge: NSObject, CAPBridgeProtocol {
         self.messageHandlerWrapper = messageHandlerWrapper
         self.config = configuration
         self.cordovaParser = cordovaConfiguration
+        self.userNotificationDelegate = UserNotificationCenterDelegate()
+
+        if configuration.assignUserNotificationCenterDelegate {
+            let center = UNUserNotificationCenter.current()
+            center.delegate = userNotificationDelegate
+        }
```

Comment:
What if this was moved into `UserNotificationCenterDelegate`? It could have a computed property that both returned whether or not it was the active delegate and allowed it be to set. The bridge, then, could just set the flag like:

`userNotificationDelegate.handleNotifications = configuration.assignUserNotificationDelegate`

Reply:
I wasn't quite sure where to place it because it's such global logic. I can change it to see how it looks.